scalafix-cli/src/main/scala/scalafix/cli/CliRunner.scala

```diff
@@ -220,11 +237,25 @@ object CliRunner {
             .notOk
         }
       }
-    val resolvedInputs: Configured[Seq[Input]] = {
-      mirrorPaths.map { fromMirror =>
-        val result = fromMirror ++ explicitPaths
-        result
-      }
+    private val resolvedPathMatcher: Configured[FilterMatcher] = try {
+      Ok(FilterMatcher(include, exclude))
+    } catch {
+      case e: PatternSyntaxException =>
+        ConfError
+          .msg(
+            s"Invalid '${e.getPattern}' for  --include/--exclude. ${e.getMessage}")
+          .notOk
+    }
+    val resolvedInputs: Configured[Seq[Input]] = for {
+      fromMirror <- mirrorPaths
+      pathMatcher <- resolvedPathMatcher
+      sourceroot <- resolvedSourceroot
+    } yield {
+      def inputOK(input: Input) = pathMatcher.matches(input.path(sourceroot))
+      val builder = Seq.newBuilder[Input]
+      fromMirror.withFilter(inputOK).foreach(builder += _)
+      explicitPaths.withFilter(inputOK).foreach(builder += _)
```

Comment:
It's still a bit awkward that there is both a list of explicit files, and a list of filters... but probably ok, because they'll be used in different contexts.